### DISCUSSION TITLE
[Feature] "Integrated libraries" for partner sharing

### DISCUSSION THREAD

**Discussion by lxrst** [immich-app/immich#5457] [Feature Request]
### The feature

An extension to partner sharing's "Show in timeline" feature that adds partners' face and object classification to the "Explore" view

### Platform

- [X] Server
- [X] Web
- [ ] Mobile

**Comment by lxrst** [comment]
I'd like to try my hand at implementing this.

**Comment by kladderadeng** [comment]
I wanted to create a new post about the issue I have currently and I am not 100% sure if this exactly what you are searching for.

After starting Immich, importing my existing Library and going through hundreds of detected faces to clean everything up, I shared this imported Library with my partner. I was surprised that Search and People was not showing up for my partner. We then found the option in the Account settings to integrate shared libraries into the timeline and thought that this might fix it. It did not.

Now my Partner have a hundred thousand pictures in the timeline without the people feature and not even the search working for it.

Currently I think each family member needs to import the same external Library, go through the hours of import + people tagging again to have the same result. And it would eat up a lot of unnecessary disk space.

So if what you proposed is, that with sharing a library, also the people tagging and search / metadata search functionality is shared, I´m all in and I would love to see this :-)

**Reply by lxrst** [reply]
Yes, that's exactly what this is. I'm kinda busy with work, so it may be at most a couple weeks until I actually have this implemented. Thanks for your support!

**Reply by kladderadeng** [reply]
Thank you for YOUR support. Really! As someone who can't code a working Hello world example, I am very grateful what people like you do in your free time to help projects like this. And Immich made leaps since I had a look at it a few months ago. Thank you very much :-)

**Comment by BlackBadPinguin** [comment]
Also just wanted to write a post about this, big thank you to you, if you need any help hit me up :)

**Comment by lxrst** [comment]
I've been meaning to write up an update on this, but haven't gotten around to it until now.
I can't find the code I was looking at before and someone can correct me if I'm wrong, but I believe the "show in timeline" feature simply pulls the partner's timeline separately client-side, combining them into one feed.
If I am right, this is a structural issue, and it'd be much more complex to implement this feature than I thought.
I don't know the codebase very well, so it's probably best if I don't try to restructure things.

**Reply by kladderadeng** [reply]
Thanks for your time looking into it and trying. This seems to be another good example where one (read: me, as a user) thinks "this can not be too complicated" unless you know the details and realize that things under the hood are not working as you thought they are :-)

**Reply by ttyridal** [reply]
Not sure how it was before the January release, but it appears to happen in the backend:
`
api/asset/time-bucket?.....&withPartners=true&withStacked=true
`
gives me all my and my partners assets.

 api/search/explore and api/person requests (from explore view) does not have that *withPartners* flag.. (and adding it manually doesn't help 🙃).  
 
might be a place to start..

But then, there will be questions with non-obvious answers. Like face thumbnails. What to do if it's from a non-shared image? or how to handle same person in the "in-shared" and local library.


Would love to see this feature included. Even, perhaps, letting the partner participate in the chores (tagging)

**Comment by bo0tzz** [comment]
Closing as a duplicate of #5089
